src/services/authService.ts

```diff
@@ -0,0 +1,133 @@
+import { createAsyncThunk } from "@reduxjs/toolkit";
+import { auth } from "firebase-config";
+import {
+  createUserWithEmailAndPassword,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+import { toast } from "react-toastify";
+import { collection, addDoc, getDocs, doc, setDoc, updateDoc } from "firebase/firestore";
+import { db } from "firebase-config";
+import { getUserCredentials } from "redux/slices/authSlice";
+
+export const addUser = (createAsyncThunk as any)(
+  "auth/addUser",
+  async ({
+    id,
+    name,
+    username,
+    email,
+  }: {
+    id: string;
+    username: string;
+    name: string;
+    email: string;
+  }) => {
+    try {
+      // **** Add doc in collection without id ****
+      // await addDoc(collection(db, "users"), {
+      //   name,
+      //   username,
+      //   email,
+      // });
```

Comment:
this seems ambiguous. you can remove commented code, if not require for further ref.